### DISCUSSION TITLE
ESP32 DISPLAY

### DISCUSSION THREAD

**Discussion by B1Gt0ne** [ArminJo/Arduino-BlueDisplay#13] [Q&A] [answered]
I cant get the display scale or size to change when using an ESP32 with my Android phone
All I can get is 51.6% 696x1080
If using an UNO I can change the screen to anything up to 2094x1080
Am I missing something? Is there something I need to change/define in one of the 'src' files?

**Comment by ArminJo** [comment]
What is the suspected function/command which does not work for ESP32?

**Comment by B1Gt0ne** [comment]
The following test sketch runs on the UNO and the ESP32.

They both start at a screen resolution of 51.6% 696x1080 but the UNO will change to full screen 100% 2094x1080 and the ESP32 stays at 51%

 

<>
#include <Arduino.h>
#define DO_NOT_NEED_BASIC_TOUCH_EVENTS // Disables basic touch events like down, move and up. Saves 620 bytes program memory and 36 bytes RAM
#include "BlueDisplay.hpp"
void initDisplay(void);
void drawGui(void);
int sCurrentDisplayWidth;
int sCurrentDisplayHeight;
#define RED ((color16_t)0xF800)
void setup() {
#if defined(ESP32)
  Serial.begin(115200);
  Serial.println("START " __FILE__ " from " __DATE__ "\r\nUsing library version " VERSION_BLUE_DISPLAY);
  initSerial("ESP-BD_Example");
  Serial.println("Start ESP32 BT-client with name \"ESP-BD_Example\"");
#else
  initSerial();
#endif
  BlueDisplay1.initCommunication(&initDisplay, &drawGui);
#if defined(USE_SERIAL1) // defined in BlueSerial.h
#if defined(__AVR_ATmega32U4__) || defined(SERIAL_PORT_USBVIRTUAL) || defined(SERIAL_USB) /*stm32duino*/|| defined(USBCON) /*STM32_stm32*/|| defined(SERIALUSB_PID) || defined(ARDUINO_attiny3217)
  delay(4000); // To be able to connect Serial monitor after reset or power up and before first print out. Do not wait for an attached Serial Monitor!
#endif
#endif
}
void loop() {
  BlueDisplay1.clearDisplay(COLOR16_RED);
  BlueDisplay1.drawText(24, 90, "TOP LEFT = 24,24", 88, COLOR16_BLACK, COLOR16_NO_BACKGROUND);
  delay(1500);
  BlueDisplay1.clearDisplay(COLOR16_BLUE);
  delay(1500);
  BlueDisplay1.fillRect(0, 0, 1047, 540 , COLOR16_RED );
  BlueDisplay1.drawRect(1000, 600, 2000, 1000, COLOR16_WHITE, 13);
  BlueDisplay1.drawCircle(1047, 540, 100, COLOR16_BLACK, 40);
  BlueDisplay1.drawLineWithThickness(1047, 540, 1347, 240, 40, COLOR16_BLACK);
  delay(1500);
}
void initDisplay(void) {
  sCurrentDisplayWidth = BlueDisplay1.getMaxDisplayWidth();
  sCurrentDisplayHeight = BlueDisplay1.getMaxDisplayHeight();
  BlueDisplay1.setFlagsAndSize(BD_FLAG_FIRST_RESET_ALL | BD_FLAG_USE_MAX_SIZE | BD_FLAG_TOUCH_BASIC_DISABLE, sCurrentDisplayWidth, sCurrentDisplayHeight);
}
void drawGui(void) {
  BlueDisplay1.clearDisplay(COLOR16_RED);
}
<>

**Comment by ArminJo** [comment, marked as answer]
Thanks, 
could you please delete the original mail content, the empty lines and mark code as code section -> see `<>` menue point above.

**Reply by B1Gt0ne** [reply]
I think I have done what  you mean

**Reply by ArminJo** [reply]
Thanks.
Please have a look at the examples I documented yesterday. Is this understandable?
```
    /*
     * Register callback handler and check for connection still established.
     * For ESP32 and after power on at other platforms, Bluetooth is just enabled here,
     * but the android app is not manually (re)connected to us, so we are definitely not connected here!
     * In this case, the periodic call of checkAndHandleEvents() in the main loop catches the connection build up message
     * from the android app at the time of manual (re)connection and in turn calls the initDisplay() and drawGui() functions.
     */
    BlueDisplay1.initCommunication(&initDisplay, &drawGui);
```

**Reply by B1Gt0ne** [reply]
It all works OK now
Many thanks for your time
